### PR TITLE
OCM-9784 | test: automate case id:72538

### PR DESCRIPTION
**Pull request by patriksuba**
Successfully tested for all HCP profiles. For profile with autoscaling enabled test will not work as specific number of subnets have to be used.

### CODE REVIEW

**Comment by yasun1** [comment]
I agree with the automation, but I'd like to leave it to the ROSA HCP QE members to double-check the checkpoints.

File: tests/e2e/test_rosacli_cluster.go [hunk 1]
```diff
@@ -1642,6 +1643,144 @@ var _ = Describe("HCP cluster creation negative testing",
 						ContainSubstring(
 							"ERR: Additional Allowed Principals is supported only for Hosted Control Planes"))
 			})
+
+		It("HCP cluster creation subnets validation - [id:72538]",
+			labels.High, labels.Runtime.Day1Negative,
+			func() {
+				if profile.ClusterConfig.Private {
```
Comment (radtriste) [inline tests/e2e/test_rosacli_cluster.go:1650]: AFAIK, this test is HCP only

File: tests/e2e/test_rosacli_cluster.go [hunk 2]
```diff
@@ -1642,6 +1643,144 @@ var _ = Describe("HCP cluster creation negative testing",
...
+					rosalCommand.DeleteFlag("--private", false)
+					rosalCommand.DeleteFlag("--private-link", false)
+				}
+				if profile.ClusterConfig.Autoscale {
+					Skip("Test do not work with autoscaling enabled")
+				}
+				log.Logger.Info(profile.Name)
+				log.Logger.Info(strings.Split(rosalCommand.GetFullCommand(), " "))
```
Comment (radtriste) [inline tests/e2e/test_rosacli_cluster.go:1658]: Can you put these as Debug ?

File: tests/e2e/test_rosacli_cluster.go [hunk 3]
```diff
@@ -1642,6 +1643,144 @@ var _ = Describe("HCP cluster creation negative testing",
...
+
+				By("Prepare a vpc for the testing")
+				vpc, err := vpc_client.PrepareVPC(vpcName, region, "", false, "")
+				Expect(err).ToNot(HaveOccurred())
+				defer vpc.DeleteVPCChain()
+
+				subnetMap, err := profilehandler.PrepareSubnets(vpc, region, []string{}, true)
+				Expect(err).ToNot(HaveOccurred())
+
+				additionalPrivateSubnet, err := vpc.CreatePrivateSubnet(availabilityZone, true)
```
Comment (radtriste) [inline tests/e2e/test_rosacli_cluster.go:1683]: I would initialize `availabilityZone` with the AZ from  the first subnet in `subnetMap`. That way you make sure you have a duplicated AZ

File: tests/e2e/test_rosacli_cluster.go (hunk 1 above)
Comment (patriksuba) [reply tests/e2e/test_rosacli_cluster.go:1653]: Yes it is. Testing group is only for HCP negative.

File: tests/e2e/test_rosacli_cluster.go [hunk 4]
```diff
@@ -1645,6 +1646,146 @@ var _ = Describe("HCP cluster creation negative testing",
 						ContainSubstring(
 							"ERR: Additional Allowed Principals is supported only for Hosted Control Planes"))
 			})
+
+		It("HCP cluster creation subnets validation - [id:72538]",
+			labels.High, labels.Runtime.Day1Negative,
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:1651]: Day1 negative cases has no profile setting.

Comment (patriksuba) [reply tests/e2e/test_rosacli_cluster.go:1651]: https://github.com/openshift/rosa/blob/master/tests/e2e/test_rosacli_cluster.go#L1431

Comment (radtriste) [reply tests/e2e/test_rosacli_cluster.go:1651]: should be fine then

File: tests/e2e/test_rosacli_cluster.go (hunk 1 above)
Comment (radtriste) [reply tests/e2e/test_rosacli_cluster.go:1653]: should be fine then

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**